### ISSUE TITLE
[no-explicit-any] Adding switches for supported options

### ISSUE THREAD

**Issue by weirdpattern** [bradzacher/eslint-plugin-typescript#254]
I'd like to propose a change to an existing rule.

How about adding extra options to the `no-explicit-any` rule?
I'm thinking switches for the multiple cases that are currently supported, but my main concern is with the generic array declaration when the array is used as a rest parameter. 

I'm not proposing any specifics and not doing any work on it just yet as I want to hear from the community before I (or who ever wants to take it) start working on it...

So, thoughts?

**Comment by bradzacher** [comment]
> multiple cases that are currently supported

There aren't any individual cases any more, thanks to the new types provided by the parser!
https://github.com/bradzacher/eslint-plugin-typescript/blob/63311f3d0aca0ad93584eef69180862b8ed25bb8/lib/rules/no-explicit-any.js#L28-L33

**Comment by weirdpattern** [comment]
Ha! I was doing this from memory... 
I probably should stop doing that as this plugin changed a lot since we moved to the latest typescript parser... anyway... unless I'm missing something (and I apologize if I am... I've been a little busy with a toddler and a new baby on the way) I think this is still a valid request, isn't it? Would be nice to have a little more control over what gets validated vs ignored.

**Comment by bradzacher** [comment]
Ha ha no problem.

I don't think that more config would be a bad thing, as long as it's easy to keep the current "no anys are okay" functionality.

Considering we cut the rule down 5 lines, it should be fine to add some complexity back :)

**Comment by armano2** [comment]
@weirdpattern we can add configuration

but we have to define, base one what we want to limit reporting, and by default all options should be enabled :)

**Comment by weirdpattern** [comment]
@armano2 I agree. I already have some options in mind, but I won't be able to work on this until after the holidays.
